### PR TITLE
efi: Improve Windows Guest Support

### PR DESCRIPTION
**Pull request by retrage**
The changes in this PR are to support Windows guests. I previously added EFI variable feature to support stock Ubuntu images, but it seems to be needed to boot Linux and blocks Windows boot. So let's opt out the feature.

### CODE REVIEW

**Comment by rbradford** [comment]
@retrage The issue you're seeing on Windows might be due to CH issues. I've created a PR to update to the latest CH and hopefully if you pick that up you will be able to make some progress.

**Comment by rbradford** [comment]
@retrage If the CI takes too long (>~30 minutes) feel free to run another VM in parallel for the Windows tests.

File: src/integration.rs [hunk 1]
```diff
@@ -325,176 +335,332 @@ mod tests {
...
+        #[test]
+        #[cfg(not(feature = "coreboot"))]
+        fn test_boot_ch_windows() {
+            let tmp_dir = TempDir::new().expect("Expect creating temporary directory to succeed");
+            let os = prepare_os_disk(&tmp_dir, WINDOWS_IMAGE_NAME);
+
+            let mut c = Command::new("./resources/cloud-hypervisor");
+            c.args(&[
+                "--cpus",
+                "boot=2,kvm_hyperv=on",
```
Comment (rbradford) [inline src/integration.rs:628]: Do you need the equivalent of `kvm_hyperv=on` for QEMU?

Comment (retrage) [reply src/integration.rs:628]: As far as I tested, no Windows-specific option is required for coreboot/QEMU, but I didn't test the Windows guest with QEMU `-kernel` option mode. I think this test combination cannot be supported for now.

**Comment by retrage** [comment]
The main blocking issues are:
1. Fetching Windows VM image takes too long (~7 min.)
2. The attached network device number vary on QEMU.

For issue 1, we can mitigate it by adding another parallel stage for the new tests.

Issue 2 is the biggest issue. Since the attached network device is automatically configured as a new device, we cannot set the static IP address at boot time. This blocks from sending ssh command from the test handler. Is it possible to configure the Windows VM using cloud-init like other Linux test targets?

**Comment by rbradford** [comment]
@retrage I'd be very content with just testing CH + Windows. We can address QEMU support later on.

**Comment by retrage** [comment]
@rbradford OK. I disabled the Windows guest tests for QEMU.

**Review by rbradford — APPROVED** [review]
You can add `#[ignore]` on tests. This means you can still explicitly run them using a filter but they won't normally run.

File: src/efi/mod.rs [hunk 2]
```diff
@@ -311,9 +319,13 @@ pub extern "win64" fn set_variable(
...
-    VARIABLES
-        .borrow_mut()
-        .set(variable_name, vendor_guid, attributes, data_size, data)
+    if cfg!(feature = "efi-var") {
+        VARIABLES
+            .borrow_mut()
+            .set(variable_name, vendor_guid, attributes, data_size, data)
+    } else {
+        Status::UNSUPPORTED
+    }
```
Comment (rbradford) [inline src/efi/mod.rs:328]: Just so i'm clear. Nothing currently requires EFI variables to boot?

**Pull request merged**